bblocks/import_tools/world_bank_projects.py

```diff
@@ -0,0 +1,559 @@
+        """Format sector data and store it as a dataframe in _data attribute
+        with key 'sector_data'"""
+
+        sector_data = []
+        for _, proj_data in self._raw_data.items():
+            if "sector" in proj_data:
+                proj_id = proj_data["id"]
+
+                sectors = _get_sector_data(proj_data)
+                sector_data.extend(
+                    [
+                        {"project ID": proj_id, "sector": s, "percent": p}
+                        for s, p in sectors.items()
+                    ]
+                )
+
+        self._data["sector_data"] = pd.DataFrame(sector_data)
+
+    def _download(self, additional_fields: list | None = None) -> None:
+        """Download data from World Bank Projects API and save it as a json file."""
+
+        logger.info(f"Starting download of World Bank Projects")
+
+        if additional_fields is None:
+            additional_fields = []
+        if isinstance(additional_fields, str):
+            additional_fields = [additional_fields]
+
+        with open(self._path, "w") as file:
+            data = (
+                QueryAPI(
+                    start_date=self.start_date,
+                    end_date=self.end_date,
+                    fields=list(GENERAL_FIELDS)
+                    + list(OTHER_FIELDS)
+                    + additional_fields,
+                )
+                .request_data()
+                .get_data()
+            )
+            json.dump(data, file)
+
+        logger.info(f"Successfully downloaded World Bank Projects")
+
+    def load_data(self, *, additional_fields: str | list = None) -> ImportData:
+        """Load data to the object
+
+        This method will load the World Bank Project data to the object.
+        If the data has already downloaded, it will be loaded to the object from disk,
+        otherwise it will be downloaded from the API and saved as a json file and  loaded
+        to the object.
+
+        Args:
+            additional_fields: additional fields to download from the API. If the data has
+                already been downloaded, the additional fields may not be loaded if they do not
+                exist in the downloaded file. To force download of data with additional fields,
+                use the update_data method passing the additional fields as argument
+
+        Returns:
+            object with loaded data
+        """
+
+        # check if additional fields is a string or None and convert to list
+        if additional_fields is None:
+            additional_fields = []
+        if isinstance(additional_fields, str):
+            additional_fields = [additional_fields]
+
+        # if file does not exist, download it and save it as a json file
+        if not self._path.exists():
+            self._download(additional_fields=additional_fields)
+
+        # if file exists and additional fields are passed, log warning
+        else:
+            if not additional_fields:
```

Comment:
This check doesn't seem to work as expected. It logs a warning when no fields are passed but data is read from the disk. Should it be `if additional_fields is not None`?